src/pin_vec.rs

```diff
@@ -36,25 +37,56 @@ impl<T> PinVec<T> {
         }
     }
 
-    pub fn get_mut(&mut self, index: usize) -> Option<&mut T> {
+    pub fn get_mut(&mut self, index: usize) -> Option<&mut T>
+    where
+        T: Unpin,
+    {
+        // Safety: get_unchecked_mut lets us ignore the Pin part, but since T is Unpin
+        // it's okay to Unpin it.
+        unsafe { self.get_unchecked_mut(index) }
```

Comment:
Minor suggestion here in how the APIs are structured.

This could be entirely safe if it used `get_pin_mut` (since `T` is `Unpin` it can use [`Pin::get_mut`](https://doc.rust-lang.org/std/pin/struct.Pin.html#method.get_mut)), and `get_pin_mut` could fully replace `get_unchecked_mut`.

In fact `get_mut` might even be superfluous since you can already achieve what it does externally if `T` is `Unpin`.

Reply:
Oh, good idea! I gave that a try and it works out pretty nicely. Combined with `pin_project`, I was able to get rid of `get_unchecked_mut`, and now `pin_slab.rs` is entirely safe code.